modules/catalog/src/main/java/org/apache/ignite/internal/catalog/CatalogServiceImpl.java

```diff
@@ -773,11 +783,21 @@ private void registerCatalog(Catalog newCatalog) {
         catalogByTs.put(newCatalog.time(), newCatalog);
     }
 
-    private CompletableFuture<Void> saveUpdate(UpdateProducer updateProducer) {
-        return saveUpdate(updateProducer, 0);
+    private CompletableFuture<Void> saveUpdateAndWaitForActivation(UpdateProducer updateProducer) {
+        return saveUpdate(updateProducer, 0)
+                .thenCompose(newVersion -> versionTracker.waitFor(newVersion)
```

Comment:
We have to wait for versionTracker because this is how we make sure that the version we just created is available in the CatalogServiceImpl internal mappings; and we need this to obtain the new version activation moment. Only when we know the activation moment can we compute the target moment we need to wait for.

The necessity to wait for `versionTracker` to get activation moment will become obsolete when https://issues.apache.org/jira/browse/IGNITE-19796 gets implemented.

Thanks, I applied your suggestion with a minor modification.

Reply:
I understand why we need to wait for the `versionTracker`, my question was why we do it here since we also wait for the `versionTracker` inside `saveUpdate`, so now it happens twice. Or am I missing something?

Reply:
Oh I missed that this wait already happens. Thanks, removed the wait